tests/integration/istioctl/wait_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioctl
+
+import (
+	"fmt"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+)
+
+func TestWait(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix:   "default",
+				Inject:   true,
+				Revision: "",
+				Labels:   nil,
+			})
+			ctx.ApplyConfigOrFail(t, ns.Name(), `
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews
+spec:
+  gateways: [missing-gw]
+  hosts:
+  - reviews
+  http:
+  - route:
+    - destination: 
+        host: reviews
+`)
+			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+			fmt.Print("start")
+			istioCtl.InvokeOrFail(t, []string{"x", "wait", "VirtualService", "reviews." + ns.Name()})
```

Comment:
Is this expected to pass or fail? Since it has a missing gw?

Reply:
Yes.  The wait command just waits for the object to be distributed or skipped.  If we had analysis turned on (which we don't), we could expect to see a PassedAnalysis=false condition, with a validationMessages field populated to complain about the missing gateway.  But none of that blocks distribution.